### ISSUE TITLE
Broken terms with Collect, TFORM -w >= 3

### ISSUE THREAD

**Issue by tueda** [vermaseren/form#11]
If I run the following program, which is basically a test example heavily using the Collect statement, with -w3 or more threads in TFORM, something wrong often happens, like "Unexpected code in ReNumber" or just freezing, probably depending on some race condition:
```
On fewerstats;

* The problem size.
#ifndef `N'
#define N "8"
#endif

#message N=`N'

S x1,...,x19;
CF f1,f2;

L F = (x1+...+x19)^`N';
.sort

* Repeat Bracket-Collect.

#do i=1,9
  B x`i',x1`i';
  .sort:B`i';
  Collect, f1, f2;
  .sort:Collect`i';
  id f1(x1?) = x1;
  id f2(x2?) = x2;
#enddo

* Check the result.

.sort
#do i=1,19
  id x`i' = 1;
#enddo
.sort
Drop F;
L ZERO = F - 19^`N';
.sort
#if `ZERO_' == 0
  #message FAILED
  #terminate 1
#endif
.sort
#message OK
.end
```
I also ran the program with -w0, -w1, -w2 more than 100 times for each case, but I haven't experienced any problems. I guess there might be a thread unsafe code around SortBot(?)

This issue was found originally in a discussion with Matthias Steinhauser.

**Comment by tueda** [comment]
**Status** : The above example now works due to the change in 1129d04ca9, which lowered the default "percentage" to 40 (less efficient but safe). Changing the "percentage" like
```
Collect, f1, f2, 51;
```
make the program crash again.

**Comment by tueda** [comment]
Fixed in 59b949e469.
